geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/Transformers.scala

```diff
@@ -117,16 +146,24 @@ object Transformers extends JavaTokenParsers {
   case class LitString(value: String) extends Lit[String]
   case class LitInt(value: Integer) extends Lit[Integer]
   case class LitLong(value: Long) extends Lit[Long]
+  case class LitFloat(value: java.lang.Float) extends Lit[java.lang.Float]
   case class LitDouble(value: java.lang.Double) extends Lit[java.lang.Double]
+  case class LitBoolean(value: java.lang.Boolean) extends Lit[java.lang.Boolean]
+
   case class Cast2Int(e: Expr) extends Expr {
     override def eval(args: Array[Any])(implicit ctx: EvaluationContext): Any = e.eval(args).asInstanceOf[String].toInt
   }
-
+  case class Cast2Long(e: Expr) extends Expr {
+    override def eval(args: Array[Any])(implicit ctx: EvaluationContext): Any = e.eval(args).asInstanceOf[String].toLong
```

Comment:
i can imagine string -> number and num -> num like double to float, double to int, or something like that. but realistically i think 99% of the time it's parsing a CSV and casting a string to an int. long term i think the converters should do type inference if they don't already. i.e. if the field is a double, it should be an implicit ::double cast without specifying it.